src/core/ddsc/tests/qos.c

```diff
@@ -240,6 +239,44 @@ qos_fini(void)
 /****************************************************************************
  * API tests
  ****************************************************************************/
+Test(ddsc_qos, copy_bad_source, .init=qos_init, .fini=qos_fini)
+{
+    dds_return_t result;
+
+	result = dds_qos_copy(g_qos, NULL);
+	cr_assert_eq(dds_err_nr(result), DDS_RETCODE_BAD_PARAMETER, "returned %d", dds_err_nr(result));
+}
+
+Test(ddsc_qos, copy_bad_destination, .init=qos_init, .fini=qos_fini)
+{
+	dds_return_t result;
+
+	result = dds_qos_copy(NULL, g_qos);
+	cr_assert_eq(dds_err_nr(result), DDS_RETCODE_BAD_PARAMETER, "returned %d", dds_err_nr(result));
+}
+
+Test(ddsc_qos, copy_with_partition, .init=qos_init, .fini=qos_fini)
+{
+	dds_return_t result;
+	dds_qos_t *qos;
+	struct pol_partition p = { 0, NULL };
+
+	qos = dds_qos_create();
+	cr_assert_not_null(qos);
+
+	dds_qset_partition(g_qos, g_pol_partition.n, c_partitions);
+	result = dds_qos_copy(qos, g_qos);
+
+	cr_assert_eq(result, DDS_RETCODE_OK);
+	dds_qget_partition(qos, &p.n, &p.ps);
+	cr_assert_eq(p.n, 2);
```

Comment:
Asserting twice than p.n is equal to the same value makes no sense; checking against g_pol_partition.n is the better one, so this one should go.